### PR TITLE
[PBI-1252] Fix issue where preroll overlay was being time out by AMC

### PR DESCRIPTION
**Pull request by dchauOoyala**
- Issue was caused by a TypeError in the fw_onSlotStarted function
- Similar issue found in fw_onSlotEnded function
- Add check to prevent TypeError in the event that adVideoElement is null or if the ad is non-linear

### CODE REVIEW

File: js/freewheel.js [hunk 1]
```diff
@@ -385,6 +386,7 @@ OO.Ads.manager(function(_, $) {
      */
     this.playAd = function(ad) {
       _resetAdState();
+      currentAd = ad;
```
Comment (angieisrani) [inline js/freewheel.js]: We already save currentPlayingSlot - are you able to use that?

Comment (dchauOoyala) [inline js/freewheel.js]: I did see the currentPlayingSlot variable. That get's assigned as ad.ad on line 407, which no longer has the isLinear field. Is there something similar to isLinear from ad.ad?

Comment (angieisrani) [inline js/freewheel.js]: Slot has getTimePositionClass() which can tell us if it's a preroll, midroll, postroll, or overlay.  Technically it does not differentiate between linear and nonlinear overlays so we assume the overlay to be nonlinear

Comment (dchauOoyala) [inline js/freewheel.js]: All right, I'll change it and use getTimePositionClass() instead

File: js/freewheel.js [hunk 2]
```diff
@@ -742,7 +742,12 @@ OO.Ads.manager(function(_, $) {
      * @method Freewheel#fw_onSlotStarted
      */
     var fw_onSlotStarted = function() {
-      amc.ui.adVideoElement.removeAttr('controls');
+      // adVideoElement may be null for overlays
+      if (currentPlayingSlot &&
+          currentPlayingSlot.getTimePositionClass() === tv.freewheel.SDK.TIME_POSITION_CLASS_OVERLAY &&
```
Comment (pilievOoyala) [inline js/freewheel.js]: isn't this supposed to check if it's a linear ad (not overlay) and then remove the attribute?

Comment (dchauOoyala) [inline js/freewheel.js]: you're right

**Comment by gfrankOoyala** [comment]
Looks good to me.

**Pull request merged**